### PR TITLE
Note on the use of DP and DDP

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Adding a note on the use of dp and ddp with metrics (based on question we got on slack)

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/pages/overview.rst [hunk 1]
```diff
@@ -109,6 +109,29 @@ the native `MetricCollection`_ module can also be used to wrap multiple metrics.
             val3 = self.metric3['accuracy'](preds, target)
             val4 = self.metric4(preds, target)
 
+Metrics in Dataparallel (DP) mode
+=================================
+
+When using metrics in `Dataparallel (DP) <https://pytorch.org/docs/stable/generated/torch.nn.DataParallel.html#torch.nn.DataParallel>`_
+mode, one should be aware DP will both create and destroy replicates of ones metric during a single forward pass.
```
Comment (Borda) [inline docs/source/pages/overview.rst:116]: maybe not "destroy", rather clean-up or delete?

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: docs/source/pages/overview.rst (hunk 1 above)
Comment (ananyahjha93) [reply docs/source/pages/overview.rst:116]: I agree with @Borda . "DP will both create and clean-up replicas of Metric objects"

File: docs/source/pages/overview.rst [hunk 2]
```diff
@@ -109,6 +109,29 @@ the native `MetricCollection`_ module can also be used to wrap multiple metrics.
...
+mode, one should be aware DP will both create and destroy replicates of ones metric during a single forward pass. 
+This has the consequence, that the metric state of the replicas will as default be destroyed before we can sync 
+them. It is therefore recommended, when using metrics in DP mode, to initialize them with ``dist_sync_on_step=True``
+such that metric states are synchonized between the main process and the replicas before they are destroyed.
+
+Metrics in Distributed Data Parallel (DDP) mode
+===============================================
+
+When using metrics in `Distributed Data Parallel (DPP) <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`_
+mode, one should be aware that DPP will add addtional samples to your dataset if the size of your dataset is
```
Comment (ananyahjha93) [inline docs/source/pages/overview.rst:125]: correction: DDP

**Review by ananyahjha93 — APPROVED** [review]
